coordinator/coordinator.go

```diff
@@ -276,24 +270,32 @@ func (c *Coordinator) WatchForNewJobs() error {
 	c.Logger.Info("Watching for new jobs")
 	// TODO: change it so we can watch for new jobs using task manager; searching for PENDING status tasks
 	// JOB__TYPE__NAME__VARIANT
-	runs, err := c.taskManager.GetAllTaskRuns()
+	runs, err := c.Metadata.Tasks.GetAllRuns()
 	if err != nil {
 		return fferr.NewInternalError(err)
 	}
-	runs.FilterByStatus(scheduling.Pending)
+
+	runs.FilterByStatus(scheduling.PENDING)
```

Comment:
This function is really weird in that it takes the existing runs obj and changes the pointer to it. If you're going to do that just return a new copy of a list in case runs is still needed elsewhere. It embraces immutability and cleaner to deal with.

```
pendingRuns := runs.FilterByStatus(scheduling.PENDING)
```

The even better approach is to do `c.Metadata.Tasks.GetRunsByStatus(...)`


A few questions:
- Should this be a set rather than a list? What happens if there are repeated runs?